core/services/assets/AssetManifest.php

```diff
@@ -0,0 +1,329 @@
+<?php
+
+namespace EventEspresso\core\services\assets;
+
+use EventEspresso\core\domain\DomainInterface;
+
+/**
+ * Class for loading parsing and retrieving data from an asset manifest file
+ *
+ * @package EventEspresso\core\services\assets
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+class AssetManifest
+{
+
+    const ASSET_EXT_CSS      = '.css';
+
+    const ASSET_EXT_JS      = '.js';
+
+    const ASSET_EXT_PHP     = '.php';
+
+    const FILE_NAME         = 'asset-manifest.json';
+
+    const KEY_DEPENDENCIES  = 'dependencies';
+
+    const KEY_ENTRY_POINTS  = 'entrypoints';
+
+    const KEY_FILES         = 'files';
+
+    const KEY_VERSION       = 'version';
+
+
+    /**
+     * @var array
+     */
+    private $asset_files;
+
+    /**
+     * @var string
+     */
+    private $assets_namespace;
+
+    /**
+     * @var string
+     */
+    private $assets_path;
+
+    /**
+     * @var DomainInterface
+     */
+    protected $domain;
+
+    /**
+     * @var array
+     */
+    private $entry_points;
+
+    /**
+     * @var array
+     */
+    private $manifest;
+
+    /**
+     * @var string
+     */
+    private $manifest_path;
+
+    /**
+     * This is a list of known handles that are used for css.
+     * @var array
+     */
+    private $wp_css_handle_dependencies = [
+        'wp-components',
+        'wp-block-editor',
+        'wp-block-library',
+        'wp-edit-post',
+        'wp-edit-widgets',
+        'wp-editor',
+        'wp-format-library',
+        'wp-list-reusable-blocks',
+        'wp-nux',
+    ];
+
+
+    /**
+     * AssetManifest constructor.
+     *
+     * @param DomainInterface $domain
+     */
+    public function __construct(DomainInterface $domain)
+    {
+        $this->domain = $domain;
+    }
+
+
+    /**
+     * @return void
+     */
+    public function initialize()
+    {
+        if (! $this->manifest) {
+            $this->assets_namespace = $this->domain->assetNamespace();
+            $assets_path = $this->domain->distributionAssetsPath();
+            $this->assets_path = trailingslashit($assets_path);
+            $this->setManifestFilepath();
+            $this->loadManifest();
+            $this->getAssetFiles();
+            $this->getEntryPoints();
+        }
+    }
+
+
+    /**
+     * @param string $manifest_path
+     */
+    public function setManifestFilepath($manifest_path = '')
+    {
+        $manifest_path = $manifest_path ? $manifest_path : $this->assets_path . AssetManifest::FILE_NAME;
+        if (! is_readable($manifest_path)) {
+            throw new AssetManifestException(
+                sprintf(
+                    esc_html__(
+                        'Manifest file was not found or is not readable. Please try running "yarn dev" in the console and/or verify that "%1$s" exists and has valid permissions.',
```

Comment:
Should we mention `"yarn dev"` in core for common users? May be ask them to contact support or refer to some docs?

Reply:
ya that's a better idea, although that exception is only thrown in WP_DEBUG is true, so that does imply that a "developer" would see it

Reply:
What in case debugging is not enabled? What would the user see?

Reply:
oh actually this exception is always thrown if there is no manifest file, because literally nothing will work if that's the case. I did update the notice, especially since you can't even run `yarn dev` on EDTR master now 😆